trakt/core.py

```diff
@@ -167,12 +176,140 @@ def oauth_auth(username, client_id=None, client_secret=None, store=False):
     return oauth.token['access_token']
 
 
+def get_device_code(client_id=None, client_secret=None):
+    """Generate a device code, used for device oauth authentication.
+
+    Trakt docs: https://trakt.docs.apiary.io/#reference/authentication-devices/device-code
+    :param client_id: Your Trakt OAuth Application's Client ID
+    :param client_secret: Your Trakt OAuth Application's Client Secret
+    :param store: Boolean flag used to determine if your trakt api auth data
+        should be stored locally on the system. Default is :const:`False` for
+        the security conscious
+    :return: Your OAuth device code.
+    """
+    global CLIENT_ID, CLIENT_SECRET, OAUTH_TOKEN
+    if client_id is None and client_secret is None:
+        client_id, client_secret = _get_client_info()
+    CLIENT_ID, CLIENT_SECRET = client_id, client_secret
+    HEADERS['trakt-api-key'] = CLIENT_ID
+
+    device_code_url = urljoin(BASE_URL, '/oauth/device/code')
+    headers = {'Content-Type': 'application/json'}
+    data = {"client_id": CLIENT_ID}
+
+    device_response = requests.post(device_code_url, json=data, headers=headers).json()
+    print('Your user code is: {user_code}, please navigate to {verification_url} to authenticate'.format(
+        user_code=device_response.get('user_code'), verification_url=device_response.get('verification_url')
+    ))
+
+    device_response['requested'] = time.time()
+    return device_response
+
+
+def get_device_token(device_code, client_id=None, client_secret=None, store=False):
+    """
+    Trakt docs: https://trakt.docs.apiary.io/#reference/authentication-devices/get-token
+    Response:
+    {
+      "access_token": "",
+      "token_type": "bearer",
+      "expires_in": 7776000,
+      "refresh_token": "",
+      "scope": "public",
+      "created_at": 1519329051
+    }
+    :return: Information regarding the authentication polling.
+    :return type: dict
+    """
+    global CLIENT_ID, CLIENT_SECRET, OAUTH_TOKEN, OAUTH_REFRESH
+    if client_id is None and client_secret is None:
+        client_id, client_secret = _get_client_info()
+    CLIENT_ID, CLIENT_SECRET = client_id, client_secret
+    HEADERS['trakt-api-key'] = CLIENT_ID
+
+    data = {
+        "code": device_code,
+        "client_id": CLIENT_ID,
+        "client_secret": CLIENT_SECRET
+    }
+
+    headers = {
+        'Content-Type': 'application/json'
+    }
+
+    response = None
+    try:
+        response = requests.post('https://api.trakt.tv/oauth/device/token', json=data, headers=headers)
+        response.raise_for_status()
+    except HTTPError as error:
+        if response and getattr(response, 'status_code ') == 400:
```

Comment:
looks like you have some trailing whitespace in the name of the `status_code` field. If we know that `response` isn't None I think it should be safe to switch this to

```python
if response and response.status_code == 400:
```